content/posts/2019-03-job-search-2-parsing.md

```diff
@@ -0,0 +1,118 @@
+---
+title: "Building Stack Overflow Job Search - Parsing Queries"
+date: 2019-03-22T00:00:00Z
+tags: [.net, jobs, stack-overflow]
+draft: true
+---
+
+In [Part 1](2019-02-job-search-1-intro) we talked about some of the shortcomings of Stack Overflow's job search and how we addressed them. In this episode we'll dive into how our parser is written.
+
+## Some Background
+
+Most people seem to get scared the moment the words lexer or parser get mentioned. I'd highly recommend watching [Rob Pike's talk](https://youtube.com/watch?v=HxaD_trXwRE) on the Go lexer and parser; it clears up a lot of misconceptions and provides a solid basis for writing a hand-rolled parser.
+
+You may be asking why we built a hand-rolled parser instead of something produced with a tool like ANTLR? Well, originally we were producing both C# and Javascript parsers for the backend and frontend of the site. We tried ANTLR and found the output produced for the front-end was absurdly large and not something we could realistically deploy on a production website. Our hand-rolled lexer & parser were substantially smaller and faster. That said, in the end, we decided not to ship a Javascript version of the parser.
```

Comment:
Not convinced this spiel about the hand-written parser is that interesting. Perhaps save it to the end? "other things we tried"